### PR TITLE
Hotfix Incorrect current status when updating a future OOO Status

### PR DESCRIPTION
**Pull request by heyrandhir**
- [x] Fix the error which was showing wrong status when updating future OOO Status.

Cause :
Instead of reading from API response we were reading from the input parameter which was causing the error. Now we read from the api response instead of input params.

**Dev Tested**
- [x] Yes
- [ ] No

**Browser**
- [x] Chrome
- [ ] Firefox
- [ ] MS Edge

[screen-capture (2).webm](https://user-images.githubusercontent.com/97341921/219104439-00772c2e-9042-4002-8695-e1c511155d00.webm)

### CODE REVIEW

**Review by rohan09-raj — COMMENTED** [review]
Hi @heyrandhir, I noticed another bug here, whenever I select today's date for OOO and submit the request and it always fails.
Also, can we please have a toast for success as well to improve the UX?

Here's a video demo for above scenario

[Screencast from 2023-02-16 11-49-22.webm](https://user-images.githubusercontent.com/78433013/219285553-ad9840b7-1491-4823-9e11-de2f8519c675.webm)

**Comment by heyrandhir** [comment]
> Hi @heyrandhir, I noticed another bug here, whenever I select today's date for OOO and submit the request and it always fails. Also, can we please have a toast for success as well to improve the UX?
> 
> Here's a video demo for above scenario

Thanks Rohan . I will work to fix this.

**Comment by heyrandhir** [comment]
> Hi @heyrandhir, I noticed another bug here, whenever I select today's date for OOO and submit the request and it always fails. Also, can we please have a toast for success as well to improve the UX?
> 
> Here's a video demo for above scenario

The cause for this bug was on the Backend we are converting the timestamp in UTC to make it timezone agnostic, The same was missing in the frontend. Created a helper function and implemented the same date conversion from frontend as well. 😇

Please review and approve

**Review by sahsisunny — DISMISSED** [review]

File: app/controllers/index.js [hunk 1]
```diff
@@ -25,28 +25,24 @@ export default class IndexController extends Controller {
...
-          `${statusText}. Status Update failed`,
-          '',
-          toastNotificationTimeoutOptions
-        );
-      }
+      })
+        .then((response) => response.json())
+        .then((responseData) => {
+          if (responseData.data.currentStatus?.state) {
+            this.status = responseData.data.currentStatus.state;
```
Comment (rohan09-raj) [inline app/controllers/index.js:39]: Can we have a toast to display the success of updation of the status?

Comment (heyrandhir) [reply app/controllers/index.js:39]: Done

**Review by rohan09-raj — APPROVED** [review]

**Pull request merged**